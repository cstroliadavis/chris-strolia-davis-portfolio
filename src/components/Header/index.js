export default function Header() {
  return (
    <header>
      <nav className="nav-bar">
        <div className="portfolio-owner">Chris Strolia-Davis</div>
        <div className="view-nav-toggle">
          <span className="line"></span>
          <span className="line"></span>
          <span className="line"></span>
        </div>
        <div className="menu-items">
          <ul>
            <li><a href="#">About me</a></li>
            <li><a href="#Portfolio">Portfolio</a></li>
            <li><a href="#Contact">Contact</a></li>
            <li><a href="#Resume">Résumé</a></li>
          </ul>
        </div>
      </nav>
    </header>
  );
}
