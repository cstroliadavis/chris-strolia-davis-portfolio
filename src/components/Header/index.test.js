import { render, cleanup } from '@testing-library/react';
import Header from './index'

afterEach(cleanup);

describe('Header', () => {
  it('renders', () => {
    render(<Header />);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Header />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('shows the developer\'s name', () => {
    const { getByText } = render(<Header />);

    expect(getByText('Chris Strolia-Davis')).toBeTruthy();
  });

  it('shows a navigation menu', () => {
    const { getByText } = render(<Header />);

    expect(getByText('About me')).toBeTruthy();
    expect(getByText('Portfolio')).toBeTruthy();
    expect(getByText('Contact')).toBeTruthy();
    expect(getByText('Résumé')).toBeTruthy();
  });
});
