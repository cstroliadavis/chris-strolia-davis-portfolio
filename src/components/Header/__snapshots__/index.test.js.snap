// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <nav
      class="nav-bar"
    >
      <div
        class="portfolio-owner"
      >
        Chris Strolia-Davis
      </div>
      <div
        class="view-nav-toggle"
      >
        <span
          class="line"
        />
        <span
          class="line"
        />
        <span
          class="line"
        />
      </div>
      <div
        class="menu-items"
      >
        <ul>
          <li>
            <a
              href="#"
            >
              About me
            </a>
          </li>
          <li>
            <a
              href="#Portfolio"
            >
              Portfolio
            </a>
          </li>
          <li>
            <a
              href="#Contact"
            >
              Contact
            </a>
          </li>
          <li>
            <a
              href="#Resume"
            >
              Résumé
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </header>
</DocumentFragment>
`;
